packages/jaeger-ui/src/components/Monitor/ServicesView/index.tsx

```diff
@@ -68,6 +75,8 @@ type TDispatchProps = {
   fetchAllServiceMetrics: (serviceName: string, query: MetricsAPIQueryParams) => void;
 };
 
+const trackSearchOperationDebounced = _debounce(searchQuery => trackSearchOperation(searchQuery), 1000);
```

Comment:
Just curious, why is it necessary to introduce a delay here? Aren't events reported async in the background anyway?

Reply:
it is important to prevent dummy information reporting if user search for some specific name. 
For  example if the target name is "my-metrics" I attempt to track and send report only once with completed searching query, but without debounce there are 10 separate report requests with partial searching query will be send. The cause of this behaviour is that the onChange handler fires every time a new character is typed into the input, that's why these 10 reports are sent, without debounce:
"m"
"my"
"my-"
...
"my-metric"
"my-metrics"

Reply:
Are you saying a pending debounce gets canceled if another one is issued? That was not my understanding of the function.

Reply:
regarding 1-st point - yes, you are right
regarding 2-nd point - the search field filters data only locally on front-end and doesn't sent any requests to the backend

Reply:
So on point 1 - if debounce merely delays, won't you still end up with the same incremental logs?